src/main/flight/imu.c

```diff
@@ -215,41 +213,93 @@ static float invSqrt(float x)
+        // groundspeedGain, calculated in `imuCalculateEstimatedAttitude`, range 0 - 10.0
+        // groundspeedGain is the primary multiplier of ez_ef
+        // In a GPS Rescue, groundspeedGain can be 0 - 4.5 depending on course over groundspeed vs velocity to home
+        // Otherwise, groundspeedGain is determined by GPS CPG groundspeed / GPS_COG_MIN_GROUNDSPEED
+
+        ez_ef *= groundspeedGain;
+
+        if (!FLIGHT_MODE(GPS_RESCUE_MODE)) {
+
+            const bool isWing = isFixedWing();
+
+            // 2. suppress ez_ef during and after yaw inputs, down to zero at 100% yaw
+            const float yawStickDeflectionInv = 1.0f - getRcDeflectionAbs(FD_YAW);
+            float stickDeflectionFactor = power5(yawStickDeflectionInv);
+            // negative peak detector with decay over a 2.5s time constant, to sustain the suppression
+            static float prev = 0.0f;
+            const float k = 0.4f * dt; // k = 0.004 at 100Hz, dt in seconds, 2.5s time constant
+            const float stickSuppression = prev + k * (stickDeflectionFactor - prev);
+            prev = fminf(stickSuppression, stickDeflectionFactor);
+
+            // 3. suppress ez_ef unless roll is centered, from 1.0 to zero if Roll is more than 12 degrees from flat
+            // this is to prevent adaptation to GPS while flying sideways, or with a significant sideways element
+            const float rollAngle = (float)attitude.values.roll;  // decidegrees
+            const float absRollAngle = fabsf(rollAngle);
+            float rollMax = isWing ? 250.0f : 120.0f; // 25 degrees for wing, 12 degrees for quad
+            // note: these value are 'educated guesses' - for quads it must be very tight
+            // for wings, which can't fly sideways, it can be wider
+            const float rollSuppression = (absRollAngle < rollMax) ? (rollMax - absRollAngle) / rollMax : 0.0f;
+
+            // 4. attenuate ez_ef by pitch angle, will be zero if flat or negative (ie flying tail first)
+            // allow faster adaptation for quads at higher pitch angles; returns 1.0 at 45 degrees
+            // but not if a wing, because they typically are flat when flying.
+            // need to test if anything special is needed for pitch with wings, for now do nothing.
+            float pitchSuppression = 1.0f;
+            if (!isWing) {
+                const float pitchAngle = (float)attitude.values.pitch; // decidegrees, negative is backwards
+                pitchSuppression = pitchAngle / 450.0f; // 1.0 at 45 degrees, 2.0 at 90 degrees
+                pitchSuppression = (pitchSuppression >= 0) ? pitchSuppression : 0.0f; // zero if flat or pitched backwards
+            }
+
+            ez_ef *= stickSuppression * rollSuppression * pitchSuppression;
+
+            // NOTE : these suppressions make sense with normal pilot inputs and normal flight
+            // They are not used in GPS Rescue, and probably should be bypassed in position hold, etc, 
+        }
```

Comment:
Can you at least move this into separate function? It is not  dependent on anything in IMU and can be easily multiplied into `groundspeedGain`